### ISSUE TITLE
[Idea] Crowd-training of unreleased Flickr landscape model

### ISSUE THREAD

**Issue by sidyakinian** [NVlabs/SPADE#17]
As you probably know, Flickr landscape dataset could not be released in this repo. But that model can draw landscapes with unbelievable quality, much higher than that of coco-stuff, due to training on 40k Flickr images; the fact that it hasn't been released is disappointing.

Some of us probably want to train it yourself. (Me for one, and also @Lokiiiiii brought this up) Typically it would cost a few thousand dollars. Thankfully, Product Hunt offers a paid subscription which basically offers $5000 AWS credits for $720: https://www.producthunt.com/ship#launch (Product Hunt Ship Pro, yearly subscription)

This gets the cost down to $720, but it's still a lot. Since a few of us are going to do the same exact thing, why don't we train the model together and share the cost? $720 split among 5 people is already $144, which is fair for such a powerful dataset.

Once we have a few people in, we can start a crowdfunding campaign, pledge funds, train the model and share it among us.

What do you think of this?

**Comment by code-de** [comment]
That's a great idea! Count me in.

**Comment by hologerry** [comment]
And me.

**Comment by banyet1** [comment]
Also count me in.

**Comment by ares97** [comment]
on what license such model would be?

**Comment by sidyakinian** [comment]
> on what license such model would be?

@ares97 Likely the same license that SPADE was released under

**Comment by sidyakinian** [comment]
Okay, looks like we have a few people. Let's form a group chat in some messenger to discuss the next steps. @code-de, @hologerry, and @banyet1, would you please email me which few messengers among these you find convenient?
- Messenger
- WhatsApp
- iMessage
- Discord
- Telegram
We can then choose the messenger everyone picked and host the group chat there. My email is on my profile page.
@harsh2204, @wasd96040501, if you decide to join, you're welcome to email me too!

**Comment by hologerry** [comment]
Telegram would be great. Same username as Github.

**Comment by code-de** [comment]
Telegram, Discord, WhatsApp work well for me - emailed you my usernames in each of them. Btw, @ares97, @Lokiiiiii, @aeti-in - you guys seem to be interested in this as well?

**Comment by genekogan** [comment]
@sidyakinian i am also interested in this. please include me!

**Comment by sidyakinian** [comment]
> @sidyakinian i am also interested in this. please include me!

@genekogan Sure! Please remember to email me or comment some messengers so that you can join.

**Comment by banyet1** [comment]
I'm using WhatsApp, just sent an email to u, please check it out.

**Comment by sidyakinian** [comment]
@hologerry, @banyet1 So you guys picked different options, Telegram and WhatsApp, could one of you use the other messenger so that we can gather in one chat? Either of those two is fine with me and @code-de

**Comment by banyet1** [comment]
I'm using telegram as well,  identical with WhatsApp.

**Comment by aman-tiwari** [comment]
I assume there are no segmentation masks for this dataset available, right? So those will also have to be made (or inferred using another network)

**Comment by sidyakinian** [comment]
@aman-tiwari Yes, we'll have to create the dataset ourselves. Thankfully, SPADE researchers used DeepLabV2 for it, which works pretty quickly. We'll just use that or something similar.

**Comment by noyoshi** [comment]
Not sure how much help I could be for actually training the network, but I would love to know if / when this happens! I made a pretty rudamentary web UI for this, and would love to be able to use the Flickr model on it. [Source code](https://github.com/noyoshi/smart-sketch) and public site: http://www.smartsketch.xyz

**Comment by samuel90** [comment]
@sidyakinian count me in as well, I just sent you an email

**Comment by sidyakinian** [comment]
@noyoshi Training will take one-two weeks.

As of being of help, the guys and I can go two ways: pull together our own GPU resources, or buy AWS. If we settle on the latter, anyone could be of help by pledging money; most likely only one of us (the most experienced one) will actually train the model.

**Comment by mingyuliutw** [comment]
As we mentioned in the GTC, we are making an online demo for everyone to play with the Flickr model (any mobile devices or desktops) and likely a standalone version for everyone with an NVIDIA GPU. Hopefully, it will not take too long.

**Comment by sidyakinian** [comment]
@mingyuliutw That's awesome! I've heard somewhere that it's going to be released in summer.

**Comment by Flova** [comment]
It's maybe a dumb question, but what's the main reason, of the missing Flickr model? Is it it's sheer size or are there any licencing issues?

**Comment by bitcoinmeetups** [comment]
Following

**Comment by sidyakinian** [comment]
@bitcoinmeetups Hi! Please email me your Telegram if you wanna join the chat

**Comment by datar-ai** [comment]
It's awesome ! count me in as well

**Comment by matthewbaykenney** [comment]
Very interested. Count me in

**Comment by banyet1** [comment]
We've accomplished Flickr datasets(41K) training last week.

**Comment by bensnell** [comment]
@banyet1 What are your plans now that the model is complete? Do you have any intention of making it available to others? I would love to test it out.

**Comment by prusnak** [comment]
@mingyuliutw Can you give us an update about the progress?

**Comment by genekogan** [comment]
Hey everyone! We've finished training the model and it's available here https://drive.google.com/open?id=1QJr5HBv8PAjJuVNB9zf8EiA6IcIVCswa
Here's a video of it in action: https://twitter.com/genekogan/status/1136261959970709504

**Comment by Seth-Park** [comment]
@genekogan Great work! 
Are there any plans to release the collected Flickr dataset?

**Comment by taki0112** [comment]
@genekogan Nice !!
Do you have any plans to release the dataset?

**Comment by mingyuliutw** [comment]
The official Flickr model is available as a web demo via https://www.nvidia.com/en-us/research/ai-playground/

Better models will likely come in the summer. Stay tuned.

**Comment by aeti-in** [comment]
Thanks, really appreciate.

**Comment by prusnak** [comment]
@mingyuliutw Is there a plan to release the actual model, not just the tool to play with it?

**Comment by huge123** [comment]
@hologerry thanks for your efforts,  could the flickr dataset be shared with us?

**Comment by huge123** [comment]
@genekogan Thanks for your efforts, could the flickr dataset be shared with us?

**Comment by genekogan** [comment]
i'm not sure if releasing the dataset would violate the licenses of the actual photos as they belong to other people. if we can release it, i have no problem with that.

**Comment by aviel08** [comment]
Well that's an interesting subject we're facing with machine learning. I
think we are not infringing any copyright law since we are not using any
piece of the dataset in its implicit form but in novel way, I'm no lawyer
of course.

**Comment by huge123** [comment]
> i'm not sure if releasing the dataset would violate the licenses of the actual photos as they belong to other people. if we can release it, i have no problem with that.

It is indeed a tough problem. What methods you used to extract the sematic layout of flickr images, you annotated some samples and trained the model on your own, or used existing model?

**Comment by aviel08** [comment]
What methods you used to extract the sematic layout of flickr images, you annotated some samples and trained the model on your own, or used existing model?

I always create my own datasets, either creating my labels or generating them from a 3D application but I know this a very specific scenario.

**Comment by prusnak** [comment]
@genekogan Flickr contains lots of photos which are licensed under the Creative Commons license. If you pick only these for training, I am pretty sure the trained model could by published under the same license again.

**Comment by huge123** [comment]
@aviel08 @genekogan   if it is permissible to share some sematic labels for the Flickr images along with the dataset/dataloader script, I just want to test the pretrained model, thanks.
